server/src/routes/posts.js

```diff
@@ -123,40 +123,110 @@ router.delete('/:id', firebaseAuthMiddleware, async (req, res) => {
 
 // Upvote a post
 router.put('/:id/upvote', firebaseAuthMiddleware, async (req, res) => {
-  const currentPost = await Post.findById(req.body.id);
+  const currentPost = await Post.findById(req.params.id);
+  const currentUser = await User.findOne({ email: req.user.email });
 
-  try {
-    if (req.body.upvote_type === 'clap') {
-      const claps = currentPost.upvotes_clap;
-
-      await Post.updateOne(
-        { _id: req.body.id },
-        { upvotes_clap: claps + (req.body.upvote === true ? 1 : -1) },
-      );
+  // Check body inputs are valid
+  if (currentPost == null) {
+    res.status(400).json({ message: 'Invalid post ID' });
+    return;
+  }
+  if (currentUser == null) {
+    res.status(400).json({ message: 'Invalid user ID' });
+    return;
+  }
```

Comment:
Could you just create a new user object for them if they are not already in the database? Something like
```
if (!currentUser) {
      currentUser = new User({
        email: req.user.email,
      });
  await currentUser.save();
} 
```

Reply:
Creating a user if its non-existent is done above in the create post endpoint, so could just be copied from there :)

Reply:
Added in latest commit :)